mqtt-client/src/main/java/org/fusesource/mqtt/client/CallbackConnection.java

```diff
@@ -179,13 +204,17 @@ public void onSuccess(Void value) {
                         send(request);
                     }
 
+                    reconnects = 0;
```

Comment:
Hi, why do reset reconnects=0 here, this will trigger a infinitive reconnect event if the setReconnectAttemptsMax=1